File: OnlineRadyoJava/app/src/main/java/com/example/onlineradyojava/data/RadioRoom.java
package com.example.onlineradyojava.data;

public class RadioRoom {

    private int id;
    private String roomName;
    private String currentMusic;
    private boolean playing;
    private int listenerCount;

    public RadioRoom(
            int id,
            String roomName,
            String currentMusic,
            boolean playing,
            int listenerCount
    ) {
        this.id = id;
        this.roomName = roomName;
        this.currentMusic = currentMusic;
        this.playing = playing;
        this.listenerCount = listenerCount;
    }

    public int getId() {
        return id;
    }

    public String getRoomName() {
        return roomName;
    }

    public String getCurrentMusic() {
        return currentMusic;
    }

    public boolean isPlaying() {
        return playing;
    }

    public int getListenerCount() {
        return listenerCount;
    }
}

File: online_radyo_flutter/lib/models/radio_room.dart
class RadioRoom {
  final int id;
  final String roomName;
  final String? currentMusic;
  final bool isPlaying;
  final int listenerCount;

  RadioRoom({
    required this.id,
    required this.roomName,
    required this.currentMusic,
    required this.isPlaying,
    required this.listenerCount,
  });

  factory RadioRoom.fromJson(Map<String, dynamic> json) {
    return RadioRoom(
      id: json["id"] ?? 0,
      roomName: json["roomName"] ?? "",
      currentMusic: json["currentMusic"],
      isPlaying: json["isPlaying"] ?? false,
      listenerCount: json["listenerCount"] ?? 0,
    );
  }
}